sql/pipelines/src/test/scala/org/apache/spark/sql/pipelines/graph/TriggeredGraphExecutionSuite.scala

```diff
@@ -0,0 +1,1016 @@
+      )
+    )
+  }
+
+  test("graph materialization with streams") {
+    val pipelineDef = new TestGraphRegistrationContext(spark) {
+      registerTable("a", query = Option(dfFlowFunc(Seq(1, 2).toDF("x"))))
+      registerTable("b", query = Option(readFlowFunc("a")))
+      registerView("c", query = readStreamFlowFunc("a"))
+      registerTable("d", query = Option(readStreamFlowFunc("c")))
+    }
+
+    val unresolvedGraph = pipelineDef.toDataflowGraph
+    val resolvedGraph = unresolvedGraph.resolve()
+    assert(resolvedGraph.flows.size == 4)
+    assert(resolvedGraph.tables.size == 3)
+    assert(resolvedGraph.views.size == 1)
+
+    val bFlow =
+      resolvedGraph.resolvedFlows.filter(_.identifier == fullyQualifiedIdentifier("b")).head
+    assert(bFlow.inputs == Set(fullyQualifiedIdentifier("a")))
+
+    val cFlow =
+      resolvedGraph.resolvedFlows
+        .filter(_.identifier == fullyQualifiedIdentifier("c", isView = true))
+        .head
+    assert(cFlow.inputs == Set(fullyQualifiedIdentifier("a")))
+
+    val dFlow =
+      resolvedGraph.resolvedFlows.filter(_.identifier == fullyQualifiedIdentifier("d")).head
+    assert(dFlow.inputs == Set(fullyQualifiedIdentifier("c", isView = true)))
+
+    val updateContext = TestPipelineUpdateContext(spark, unresolvedGraph)
+    updateContext.pipelineExecution.runPipeline()
+    updateContext.pipelineExecution.awaitCompletion()
+
+    assertFlowProgressStatusInOrder(
+      eventBuffer = updateContext.eventBuffer,
+      identifier = fullyQualifiedIdentifier("a"),
+      expectedFlowProgressStatus = Seq(
+        (EventLevel.INFO, FlowStatus.QUEUED),
+        (EventLevel.INFO, FlowStatus.PLANNING),
+        (EventLevel.INFO, FlowStatus.STARTING),
+        (EventLevel.INFO, FlowStatus.RUNNING),
+        (EventLevel.INFO, FlowStatus.COMPLETED)
+      )
+    )
+    assertFlowProgressStatusInOrder(
+      eventBuffer = updateContext.eventBuffer,
+      identifier = fullyQualifiedIdentifier("b"),
+      expectedFlowProgressStatus = Seq(
+        (EventLevel.INFO, FlowStatus.QUEUED),
+        (EventLevel.INFO, FlowStatus.PLANNING),
+        (EventLevel.INFO, FlowStatus.STARTING),
+        (EventLevel.INFO, FlowStatus.RUNNING),
+        (EventLevel.INFO, FlowStatus.COMPLETED)
+      )
+    )
+    assertFlowProgressStatusInOrder(
+      eventBuffer = updateContext.eventBuffer,
+      identifier = fullyQualifiedIdentifier("d"),
+      expectedFlowProgressStatus = Seq(
+        (EventLevel.INFO, FlowStatus.QUEUED),
+        (EventLevel.INFO, FlowStatus.STARTING),
+        (EventLevel.INFO, FlowStatus.RUNNING),
+        (EventLevel.INFO, FlowStatus.COMPLETED)
+      )
+    )
+
+    // no flow progress event for c, as it is a temporary view
+    assertNoFlowProgressEvent(
+      eventBuffer = updateContext.eventBuffer,
+      identifier = fullyQualifiedIdentifier("c", isView = true),
+      flowStatus = FlowStatus.STARTING
+    )
+    assert(spark.table(s"${fullyQualifiedIdentifier("d")}").count() == 2)
```

Comment:
Instead of checking the count, shall we check the data with `checkAnswer` method instead?